packages/editor/src/core/components/hooks/options.ts

```diff
@@ -26,6 +26,15 @@ export const OptionsContext = createContext<Options>({
  */
 export const useOptions = () => useContext(OptionsContext);
 
+export const UiTranslatorContext = createContext<UiTranslator>({
+  t: (label) => label,
+});
+
+/**
+ * @returns the an object with a single `t` function for ui translations
+ */
+export const useUiTranslator = () => useContext(UiTranslatorContext);
```

Comment:
maybe this would work?

```tsx
useUiTranslator = () => {
  
  const userTranslatorFunciton =  useContext(UiTranslatorContext);
  return (key: string) => {
       return userTranslatorFunciton(key) ?? key
   }
}
````


```

Reply:
@rexebin what do you think?